File: Source/ses_io/src/wrappers/fortran/wrap_array_size_next.c

#include "ses_defines.h"
#include "ses_globals.h"
#include "ses_internals.h"
#include "ses_externs.h"

#include "stdio.h"
#include <stdlib.h>

#include <stdio.h>

#undef DEBUG_WRAP

#ifdef LC_UNDER
ses_number ses_array_size_next_(ses_file_handle* pt_handle) {
#else

#ifdef UC_UNDER
ses_number SES_ARRAY_SIZE_NEXT_(ses_file_handle* pt_handle) {
#else

#ifdef LC_NOUNDER
ses_number ses_array_size_next(ses_file_handle* pt_handle) {
#else

#ifdef UC_NOUNDER
ses_number SES_ARRAY_SIZE_NEXT(ses_file_handle* pt_handle) {
#endif

#endif
#endif
#endif

  ses_number return_value = 0;

  ses_file_handle the_handle = *pt_handle;
#ifdef DEBUG_WRAP
  printf("wrap_array_size_next.c:  the_handle is %d\n", the_handle);
#endif

  return_value = ses_array_size_next(the_handle);
#ifdef DEBUG_WRAP 
  printf("wrap_array_size_next.c:  return_value is %d\n", return_value);
#endif
  return return_value;

}

 

File: Source/ses_io/src/wrappers/fortran/wrap_array_size_next.f90


#undef DEBUG_WRAP

SES_NUMBER function ses_array_size_next_f90(the_handle)

   implicit none

   SES_FILE_HANDLE, INTENT(IN) :: the_handle
   SES_NUMBER :: return_value, ses_array_size_next


#ifdef DEBUG_32BIT
   integer(kind = 4) :: return_4
   integer(kind = 8) :: return_8
#endif
#ifdef DEBUG_WRAP
   print *, "ses_array_size_next_f90:  the_handle is ", the_handle
#endif


#ifdef DEBUG_32BIT

   !print *, "debug_32bit defined"
   return_4 = ses_array_size_next(the_handle);
#ifdef DEBUG_WRAP
   print *, "return_4 is ", return_4
#endif
   return_value = return_4

#else

   !print *, "debug_32bit not defined"
   return_value = ses_array_size_next(the_handle)

#endif

#ifdef DEBUG_WRAP
   print *, "ses_array_size_next_f90:  return_value is ", return_value
#endif

   ses_array_size_next_f90 = return_value

 
end function ses_array_size_next_f90


